tests/generate_test_data.sh

```diff
@@ -3,23 +3,67 @@
 set -e
 
 
-# Regenerate rollouts and policies
-experiments/train_experts.sh -r
+TEMP=$(getopt -o '' -l all -- $@)
+if [[ $? != 0 ]]; then exit 1; fi
+eval set -- "$TEMP"
 
+while true; do
+  case "$1" in
+    --all)
+      # If `--all` option is provided then regenerate CartPole and Pendulum
+      # PPO2 experts via `train_experts.sh -r`. By default we skip this
+      # step because it takes several minutes to generate good experts.
+      #
+      # We need good (rather than quickly generated dummy) experts because
+      # `tests/test_{bc,density_baselines}.py` require particular performance
+      # measures.
+      ALL_MODE=true
+      shift
+      ;;
+    --)
+      shift
+      break
+      ;;
+    *)
+      echo "Unrecognized argument $1"
+      exit 1
+      ;;
+  esac
+done
 
-# Regenerate tests/data/imit_benchmark (for analyze.analyze_imitation tests).
-save_dir=tests/data/imit_benchmark
-if [[ -d ${save_dir} ]]; then
-  rm -r ${save_dir}
+
+sacred_dir=tests/data/imit_benchmark_sacred
+reward_models_dir=tests/data/reward_models
+SAVE_DIRS=($sacred_dir $reward_models_dir)
+if $ALL_MODE; then
+  SAVE_DIRS+=('tests/data/expert_models')
 fi
-mkdir -p ${save_dir}
 
-tmp_dir="$(mktemp -d)"
+# Wipe directories for writing later.
+for save_dir in ${SAVE_DIRS}; do
+  if [[ -d ${save_dir} ]]; then
+    rm -r ${save_dir}
+  fi
+  mkdir -p ${save_dir}
+done
+
 
-alias imit_benchmark="experiments/imit_benchmark.sh -f --log_root ${tmp_dir} \
-  --file_storage=${save_dir}"
+# Regenerate Cartpole and Pendulum expert rollouts and policies
+# (requires --all option).
+if $ALL_MODE; then
+  experiments/train_experts.sh -r
+else
+  echo "Skipping CartPole and Pendulum expert generation..."
+fi
 
+
+# Regenerate tests/data/imit_benchmark_sacred (for analysis tests)
+# and tests/data/reward_models (for `experiments/transfer_learn_benchmark.sh -f`).
+alias imit_benchmark="experiments/imit_benchmark.sh -f --file_storage=${sacred_dir}"
 shopt -s expand_aliases
-imit_benchmark --airl --run_name FOO
-imit_benchmark --gail --run_name BAR
-imit_benchmark --gail --run_name BAR
+
+imit_benchmark --gail --run_name BAR --log_root "${reward_models_dir}/gail"
+# Only need one reward model, so leave the rest in temporary directories.
+# The sacred logs are still saved to ${sacred_dir} and used for analysis test.
+imit_benchmark --gail --run_name BAR --log_root "$(mktemp -d)"
```

Comment:
Why do we run this twice (once with temp dir, once permanent directory)?

Reply:
In `test_scripts`, `test_analyze_imitation` expects `${sacred_dir}` to contain sacred logs from 2 runs of `imit_benchmark --fast --run_name BAR`.

I'm been considering somehow moving this test data generation to `test/test_scripts.py` itself. In addition to grouping all the logic in one place, it would also leave me with fewer test files to generate and commit when make changes.

Reply:
`experiments/imit_benchmark.sh -f` expects to find Cartpole and Pendulum reward models in `${reward_models_dir}/gail`. Don't need to generate this twice, so I discard the reward model from the second call in a temporary directory.

Reply:
I removed the (confusing) repeated commands by generating Sacred logs within the relevant test.